pkgs/development/python-modules/arelle/test-cli.nix

```diff
@@ -0,0 +1,119 @@
+# Run with:
+#   cd nixpkgs
+#   nix-build -A python3Packages.arelle.passthru.tests.cli
+#
+# Note: These are uninteresting generated smoke tests to verify basic functionality
+{
+  lib,
+  runCommand,
+  arelle,
+}:
+
+runCommand "arelle-test-cli${lib.optionalString (!arelle.hasGUI) "-headless"}"
+  {
+    nativeBuildInputs = [ arelle ];
+  }
+  ''
+    # Set up temporary home directory
+    export HOME=$(mktemp -d)
+
+    # Test basic CLI commands work with proper assertions
+    arelleCmdLine --version --disablePersistentConfig 2>&1 | grep "Arelle(r) 2.37.59" > /dev/null
+    arelleCmdLine --help --disablePersistentConfig 2>&1 | grep "Usage: arelleCmdLine \[options\]" > /dev/null
+    arelleCmdLine --about --disablePersistentConfig 2>&1 | grep "An open source XBRL platform" > /dev/null
+
+    ${lib.optionalString arelle.hasGUI ''
+      # Test executables exist and are accessible
+      if [ -f "${lib.getBin arelle}/bin/arelleGUI" ]; then
+        test -x "${lib.getBin arelle}/bin/arelleGUI"
+      fi
+    ''}
```

Comment:
I've made ofborg discover both `arelle` and `arelle-headless` tests on `arelle` and I've kept this for the GUI package, because we don't have a real GUI test yet.
It basically tests our `gui` parameter handling a bit.

It's the only remaining executable bit check.